frontend/src/concepts/analyticsTracking/useSegmentTracking.ts

```diff
@@ -10,28 +11,27 @@ export const useSegmentTracking = (): void => {
   const { dashboardConfig } = useAppContext();
   const username = useAppSelector((state) => state.user);
   const clusterID = useAppSelector((state) => state.clusterID);
+  const [userProps, uPropsLoaded] = useTrackUser(username);
 
   React.useEffect(() => {
-    if (segmentKey && loaded && !loadError && username && clusterID) {
-      const computeUserId = async () => {
-        const anonymousIDBuffer = await crypto.subtle.digest(
-          'SHA-1',
-          new TextEncoder().encode(username),
-        );
-        const anonymousIDArray = Array.from(new Uint8Array(anonymousIDBuffer));
-        return anonymousIDArray.map((b) => b.toString(16).padStart(2, '0')).join('');
-      };
-
+    if (segmentKey && loaded && !loadError && username && clusterID && uPropsLoaded) {
       window.clusterID = clusterID;
       initSegment({
         segmentKey,
         enabled: !dashboardConfig.spec.dashboardConfig.disableTracking,
       }).then(() => {
-        computeUserId().then((userId) => {
-          fireIdentifyEvent({ anonymousID: userId });
-          firePageEvent();
-        });
+        fireIdentifyEvent(userProps);
+        firePageEvent();
       });
     }
-  }, [clusterID, loadError, loaded, segmentKey, username, dashboardConfig]);
+  }, [
+    clusterID,
+    loadError,
+    loaded,
+    segmentKey,
+    username,
+    dashboardConfig,
```

Comment:
Remove `dashboardConfig` from the list of dependencies because it's too broad.
Assign `dashboardConfig.spec.dashboardConfig.disableTracking` to a local variable and use it as a dependency instead.